pkg/dataloaders/cedar_system_loader.go

```diff
@@ -0,0 +1,74 @@
+package dataloaders
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/graph-gophers/dataloader"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+const cedar_system_loader_key = "id"
+
+func (loaders *DataLoaders) getCedarSystemBatchFunction(ctx context.Context, keys dataloader.Keys) []*dataloader.Result {
+	logger := appcontext.ZLogger(ctx)
+	cedarSystems, err := loaders.GetCedarSystems(ctx)
+	if err != nil {
+		logger.Error("error getting cedar systems in data loader", zap.Error(err))
+	}
+
+	systemSummaryMap := make(map[string]*models.CedarSystem)
+	for _, sys := range cedarSystems {
+		if sys != nil {
```

Comment:
is it possible to run into a `nil` system?

Reply:
I think this just covers the swagger client being unable to unmarshal a system? It's also in GetSystem in the core client.